spyder/widgets/tests/test_status.py

```diff
@@ -40,5 +42,51 @@ def test_status_bar(status_bar, qtbot):
     assert len(swidgets) == 2
 
 
+class StatusBarWidgetTest(StatusBarWidget):
+    def get_tooltip(self):
+        return 'tooltip'
+
+    def get_icon(self):
+        return 'icon'
+
+
+def test_status_bar_widget_signal(status_bar, qtbot):
+    win, statusbar = status_bar
+    w = StatusBarWidgetTest(win, statusbar)
+
+    with qtbot.waitSignal(w.sig_clicked, timeout=1000):
+        qtbot.mouseRelease(w, Qt.LeftButton)
+
+    assert w.get_tooltip() == 'tooltip'
+    assert w.get_icon() == 'icon'
+
+
+def test_status_bar_conda_status(status_bar, qtbot, mocker):
+    mocker.patch.object(utils, 'is_anaconda', return_value=True)
+
+    win, statusbar = status_bar
+    w = CondaStatus(win, statusbar)
+
+    interpreter = '/miniconda/bin/python'
+    w.update_interpreter(interpreter)
+    assert w.get_tooltip() == interpreter
+    assert 'base' in w._process_conda_env_info()
+
+    interpreter = '/miniconda/envs/foo/bin/python'
+    w.update_interpreter(interpreter)
+    assert w.get_tooltip() == interpreter
+    assert 'foo' in w._process_conda_env_info()
+
+
+def test_status_bar_no_conda_status(status_bar, qtbot, mocker):
```

Comment:
I guess this test will also fail on Windows. Please take a look at that or skip it (I really have no problem with skipping it, but it's your choice).

Reply:
This one works fine since we are checking the widget is basically "empty" but I added a code in the test to make it more clear.